internal/discord/commands.go

```diff
@@ -1,11 +1,36 @@
 package discord
 
 import (
+	"fmt"
+
 	"github.com/bwmarrin/discordgo"
 )
 
 func (d *Discord) Kick(s *discordgo.Session, i *discordgo.InteractionCreate) {
-
+	options := i.ApplicationCommandData().Options
+	optionMap := make(map[string]*discordgo.ApplicationCommandInteractionDataOption, len(options))
+	for _, opt := range options {
+		optionMap[opt.Name] = opt
+	}
+
+	userToKick := optionMap["user"].UserValue(nil).ID
+
+	var err error
+	if optionMap["reason"] != nil {
+		err = d.Session.GuildMemberDeleteWithReason(i.GuildID, userToKick, optionMap["reason"].StringValue())
+	} else {
+		err = d.Session.GuildMemberDelete(i.GuildID, userToKick)
+	}
+
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not kick user <@%v>", userToKick)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		StartInteraction(s, i.Interaction, tempstr)
+	} else {
+		tempstr := fmt.Sprintf("User <@%v> has been kicked.", userToKick)
+		fmt.Printf("%v\n", tempstr)
+		StartInteraction(s, i.Interaction, tempstr)
```

Comment:
ultimately we would also want to DM the user why they are being kicked. not sure if we can do it now, but it would be nice if we're able to support that from the get-go

Reply:
ideally we should catch the error message then just ban them without giving them the reason. we would still have to log that to the channel